aws/src/main/java/org/apache/iceberg/aws/dynamodb/DynamoDbTableOperations.java

```diff
@@ -105,6 +106,8 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
     boolean newTable = base == null;
     String newMetadataLocation = writeNewMetadataIfRequired(newTable, metadata);
     CommitStatus commitStatus = CommitStatus.FAILURE;
+    AwsClientFactories.InternalRetryDetector retryDetector =
```

Comment:
I don't think we can use it as an instance variable? Since the boolean flag will not be reset, I think we need to create an new instance every execution